src/Documentation/tutorials_and_samples/tutorial_1_helloworld.md

```diff
@@ -0,0 +1,142 @@
+
+After loading the configurations, the SiloHost is built and then started asynchronously.
+
+## Configuring the client
+
+Similar to the silo, the client is configured via `ClientBuilder` and a similar collection of option classes.
+
+```csharp
+        private static async Task<IClusterClient> StartClientWithRetries()
+        {
+            attempt = 0;
+            IClusterClient client;
+            client = new ClientBuilder()
+                .UseLocalhostClustering()
+                .Configure<ClusterOptions>(options =>
+                {
+                    options.ClusterId = "dev";
+                    options.ServiceId = "HelloWorldApp";
+                })
+                .ConfigureLogging(logging => logging.AddConsole())
+                .Build();
+
+            await client.Connect(RetryFilter);
+            Console.WriteLine("Client successfully connect to silo host");
+            return client;
+        }
+
+```
+
+| Option | Used for |
+|-------------|----------|
+| `.UseLocalhostClustering()` | Same as for SiloHost |
+| `ClusterOptions` | Same as for SiloHost |
+
+A more in-depth guide to configuring your client can be found [in the Client Configuration section of the Configuration Guide.](http://dotnet.github.io/orleans/Documentation/clusters_and_clients/configuration_guide/client_configuration.html)
+
+## Writing a grain
+
+Grains are the building blocks of an Orleans application, and you can read more about them in the [Core Concepts section of the Orleans documentation.](http://dotnet.github.io/orleans/Documentation/core_concepts/index.html)
+
+This is the main body of code for the Hello World grain:
+
+```csharp
+[...]
+namespace HelloWorld.Grains
+{
+    public class HelloGrain : Orleans.Grain, IHello
+    {
+        Task<string> IHello.SayHello(string greeting)
+        {
+            logger.LogInformation($"SayHello message received: greeting = '{greeting}'");
+            return Task.FromResult($"You said: '{greeting}', I say: Hello!");
+        }
+    }
+}
+```
+
+A grain class implements one or more grain interfaces, as you can read [here, in the Grains section.](http://dotnet.github.io/orleans/Documentation/grains/index.html))
+
+```csharp
+[...]
+namespace HelloWorld.Interfaces
+{
+    public interface IHello : Orleans.IGrainWithIntegerKey
+    {
+        Task<string> SayHello(string greeting);
+    }
+}
+
+```
+
+
+
+## How the parts work together
+
+SiloHost is started first.
+Then, OrleansClient is started.
+OrleansClient creates a reference to the IHello grain and calls its SayHello() method through its interface, IHello.
+This programming model is built as part of our core concept of distributed Object Oriented Programming.
+Next, the grain is activated in the silo.
+OrleansClient sends a greeting to the activated grain.
+The grain returns a response to OrleansClient, which OrleansClient displays on the console.
+
+## Create the project structure
```

Comment:
I don't think we need this section, the sample code is enought for users to figure out this. Maybe change this section to "How ro run the sample", and link the [readme file](https://github.com/dotnet/orleans/blob/master/Samples/2.1/HelloWorld/Readme.md) in the sample.

Reply:
Yes, I will do this, but at this point, this document is an overview for the sample app, and not a collection of steps, like a "traditional" tutorial..